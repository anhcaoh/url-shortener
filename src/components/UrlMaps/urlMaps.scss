#recent-short-urls { 
    position: absolute;
    right: 0;
    top: 0;
    background: #fbfbfbe5;
    padding: 0.5rem;
    height: calc(100% - 1rem);
    border-left: 1px solid #efefef;
    transition: right 0.25s ease-in;
    &.hidden{
        right: -100%;
        transition: right 0.25s ease-in;
    }
}
#close-all-recent{
    position: absolute;
    bottom: 1rem;
    right: 1rem;
}
#open-all-recent{
    position: absolute;
    right: 1rem;
    bottom: 1rem;
}

#recent-short-urls-list {
    padding-left: 0.444rem;
    color: #e1e1e1;
    li { 
        margin-bottom: 0.5rem;
    }
}