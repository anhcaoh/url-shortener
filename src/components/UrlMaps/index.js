import React, { useState } from "react";
import { H3, Block, Text, List } from "Components/Typography";
import Link from "Components/Link";
import Button from "Components/Button";
import "./urlMaps.scss";
const UrlMaps = ( {urlMaps, reset} ) => {
    const [ isHiddenMenu, toggleHiddenMenu ] = useState( true );
    return (<>
        {(urlMaps && urlMaps.length &&
        (<><Block id="recent-short-urls" className={ isHiddenMenu ? "hidden" : "" }>
                <H3><Text>Recent Short URLs:</Text></H3>
                <List id="recent-short-urls-list">
                {urlMaps.reverse().map( (url) => (
                <li><Link href={url.shortUrl} 
                title={url.originalUrl}>{url.shortUrl}</Link></li>
                ))}</List>
                <Button id="clear-all-recent" 
                onClick={reset}>
                <Text>Clear All</Text></Button>
                <Button id="close-all-recent" 
                onClick={(e) => toggleHiddenMenu(!isHiddenMenu)}>
                <Text>Close</Text></Button>
            </Block>
            { isHiddenMenu && (<Button id="open-all-recent" 
            onClick={(e) => toggleHiddenMenu(!isHiddenMenu)}>
            <Text>Recent Short URLs</Text></Button>)}
            </>)) || null
    }</>);
};
export default UrlMaps;