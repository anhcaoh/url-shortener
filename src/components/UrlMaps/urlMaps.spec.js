import React from "react";
import renderer from "react-test-renderer";
import UrlMaps from "./index";
 
describe("UrlMaps", () => {
    it("should render snapshot", () => {
            const component = renderer.create(<UrlMaps />);
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot(); 
    });
});