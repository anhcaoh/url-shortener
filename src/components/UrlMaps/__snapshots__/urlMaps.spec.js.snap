// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlMaps should render snapshot 1`] = `null`;
